c_src/workitems.cc

```diff
@@ -767,17 +769,50 @@ work_result RangeScanTask::operator()()
 	    // If the filter has not yet been parsed, check the data
 	    // types of all fields in this key, then parse the filter,
 	    // so we know which templatized versions of the
-	    // ExpressionNodes to use
-	    //------------------------------------------------------------
+	    // ExpressionNodes to use.  If the data can't be decoded,
+	    // set filter_passed to false to ignore this key.
+	    // ------------------------------------------------------------
 
 	    if(!options_.extractor_->typesParsed_) {
-	      options_.extractor_->parseRiakObjectTypes(value.data(), value.size());
 
-	      options_.range_filter_ = 
-		parse_range_filter_opts(options_.env_, 
-                                        options_.rangeFilterSpec_, *(options_.extractor_),
-                                        throwIfBadFilter);
-	    }
+                //------------------------------------------------------------
+                // Only attempt to parse the data if it is correctly
+                // formatted, otherwise we would throw an error. 
+                //------------------------------------------------------------
+
+                if(options_.extractor_->riakObjectContentsCanBeParsed(value.data(), value.size())) {
```

Comment:
If this condition is almost invariably be true (all data are time series data and hence satisfying the condition), then it would make sense to signal this to gcc by wrapping in it the `likely` macro.